### PR TITLE
(DOCSP-13673, DOCSP-13899, DOCSP-13900) Added edit doc feature

### PR DESCRIPTION
**Pull request by atsansone**
Alena, I have the following changes for your review:

## [DOCSP-13673](https://jira.mongodb.org/browse/DOCSP-13673)

I added a [procedure](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-13673/read-document-playground#read-one-document) to the **Read Documents** page. 

  image

    - [Code for placing the added procedure](https://github.com/mongodb/docs-mongodb-vscode/pull/38/files#diff-12bbd290384cf4e81c7e3692ae31b9f8ce8d32ed5213b621c269f4110d394524R64-R66)
    - [Code for the added procedure](https://github.com/mongodb/docs-mongodb-vscode/pull/38/files#diff-13df52770094de6bb3ebe7eedf5e4965c94077e51dc072b50c1284e197fba28b)

## [DOCSP-13899](https://jira.mongodb.org/browse/DOCSP-13899)

I added that [same procedure](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-13673/databases-collections#documents) to the **Navigate Your Data** Page.

    image
  
    - [Code for added procedure](https://github.com/mongodb/docs-mongodb-vscode/pull/38/files#diff-2a81cb9e5821a67812fb6ae9b975e1d46a71a38d6ac783a4c829f3e1aa3d5732R60)

## [DOCSP-13900](https://jira.mongodb.org/browse/DOCSP-13900)

I added a [note](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-13673/databases-collections#databases-and-collections) that closing VSCode closes all open documents.

    image

    - [Code where note is added](https://github.com/mongodb/docs-mongodb-vscode/pull/38/files#diff-2a81cb9e5821a67812fb6ae9b975e1d46a71a38d6ac783a4c829f3e1aa3d5732R42)
    - [Code for new note](https://github.com/mongodb/docs-mongodb-vscode/pull/38/files#diff-58aa552db8f3a3118bbaa83faff4cac22a809f7a239beeb0c720675bfdb9bfc1)

---
[Builds without errors on `master`.](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=6000c548c5bd3658dafa63ea)

### CODE REVIEW

File: source/databases-collections.txt [hunk 1]
```diff
@@ -34,38 +34,35 @@ use the left navigation to:
...
+
 View Collection Documents and Schema
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Documents
 ``````````
 
-When you expand a collection, |vsce| displays that collection's 
-document count next to the :guilabel:`Documents` label in the 
+When you expand a collection, |vsce| displays that collection's
```
Comment (JuliaMongo) [inline source/databases-collections.txt:50]: |vsce| shows the number of documents next to the :guilabel:`Documents` label in the ...

File: source/databases-collections.txt [hunk 2]
```diff
@@ -34,38 +34,35 @@ use the left navigation to:
 Databases and Collections
 -------------------------
 
-When you expand an active connection, |vsce| shows the databases in that
-deployment. Click a database to view the collections it contains.
+When you expand an active connection, |vsce| shows the databases in
```
Comment (JuliaMongo) [inline source/databases-collections.txt:37]: [suggestion] You could turn these two sentences into an unordered list:

- Click a database to view the collections it contains.
- Click a collection to view its documents, schema, and indexes.

File: source/databases-collections.txt [hunk 3]
```diff
@@ -34,38 +34,35 @@ use the left navigation to:
 Databases and Collections
```
Comment (JuliaMongo) [inline source/databases-collections.txt:34]: [Fix] Maybe, this should be "View Databased and Collections", to make it parallel with the heading below on this same page?
I feel like you left this like that to make it an easy editorial catch :)

File: source/databases-collections.txt [hunk 4]
```diff
@@ -34,38 +34,35 @@ use the left navigation to:
...
-deployment. Click a database to view the collections it contains.
+When you expand an active connection, |vsce| shows the databases in
+that deployment. Click a database to view the collections it contains.
 
 Click a collection to view its documents, schema, and indexes.
 
+.. include:: /includes/admonitions/close-all-docs-on-exit.rst
+
 View Collection Documents and Schema
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (JuliaMongo) [inline source/databases-collections.txt:46]: [Suggestion]
I would move the sentence below in this comment and place it right under the "View Collection Documents and Schema" heading. I would edit the sentence a bit too:

When you expand a collection, |vsce| displays the number of documents it found next to the :guilabel:`Documents` label in the navigation panel.

**Comment by JuliaMongo** [comment]
I also have a few general comments.
1 The topic `databases-and-collections` is in the "Navigate your data" section. Why does it have editing content in it? Shouldn't editing belong to CRUD section below, and only there?
2 The viewing of index is not announced from the top of this topic. 
3 There are no anchor link directives of the type:  _link-to-me above subsections and the topmost intro does not have live links to these subsections but it could be useful.
4 The title does not have any Gerunds or Verbs. Does it mean: "View Databases and Collections"? Where are indexes and documents in the title?
5 In the topic that has the procedure, there is no stem sentence and instead a sentence that reads: "You may...."
6 The steps may need a bit of tightening to start with active verbs for actions, and then with info for how to make the actions in the UI. 
7 The sentence about viewing of documents within arrays is out of place. It must be on top of the section, and must be parallel to how view other documents (not in arrays). 
8 The results of the step should have a successful outcome stated first (consider reversing bullets in the result).
9 Viewing the schema does not have the verb View in the title, and needs a simple procedure, or a new para that starts with "To view, do this..." Same for the index section.

File: source/databases-collections.txt [hunk 5]
```diff
@@ -99,14 +96,14 @@ designating its type. Index key types include:
...
-   :icon-fa4:`plus` icon that appears.
+   You can open a :ref:`MongoDB Playground <vsce-crud>` pre-configured
+   to create an index by hovering over the :guilabel:`Indexes` label
+   in the navigation panel and clicking the :icon-fa4:`plus` icon that
+   appears.
 
 .. seealso::
 
-   For more information about MongoDB indexes, see the documentation in
+   To learn more about MongoDB indexes, see the documentation in
```
Comment (JuliaMongo) [inline source/databases-collections.txt:106]: [suggestion] do we need to say "the documentation in the {abc}"? Can we just say: To learn more, see <link>?

**Review by JuliaMongo — CHANGES_REQUESTED** [review]
Hi Tony, humbly submitting my review! Thank you for looking into my comments!

File: source/databases-collections.txt (hunk 1 above)
Comment (atsansone) [reply source/databases-collections.txt:50]: Agreed. Fixed.

File: source/databases-collections.txt (hunk 3 above)
Comment (atsansone) [reply source/databases-collections.txt:34]: Agreed. Fixed.

**Comment by atsansone** [comment]
@JuliaMongo : All good points, but outside the scope of the tickets / PR. Open to creating a separate Jira ticket for it.

**Comment by JuliaMongo** [comment]
@atsansone for the comment we discussed, I opened [DOCSP-13989](https://jira.mongodb.org/browse/DOCSP-13989)

**Comment by alenakhineika** [comment]
> You may edit the returned JSON document.

You can edit not all JSON documents returned by a playground, but only those that are results of `find` or `findOne` operations. Other results won't have `edit document` code lenses.

**Comment by alenakhineika** [comment]
LGTM

File: source/databases-collections.txt [hunk 6]
```diff
@@ -17,7 +17,7 @@ Navigate Your Data
 Once you :ref:`connect to your deployment <vsce-connect>` using |vsce|,
 use the left navigation to:
 
-- Explore your databases, collections, read-only views, and documents. 
+- Explore your databases, collections, read-only views, and documents.
```
Comment (JuliaMongo) [inline source/databases-collections.txt:20]: What about indexes? There is a subsection about indexes in this topic, and it does not surface in the TOC or titles shown in the TOC, just yet. Can indexes be added to this list?

File: source/includes/seq-edit-document.rst [hunk 7]
```diff
@@ -0,0 +1,9 @@
+1. At the top of this document, click :guilabel:`Edit Document`.
+   |vsce| opens it as an editable |ejson| document titled ``<database>.<collection>:"<_id>".json``.
+#. Make any edits you require.
+#. Press :kbd:`Ctrl + S` (Windows/Linux) or :kbd:`Cmd + S` to save the
+   edited document to the MongoDB database.
+
+   - If the update results in an error, |vsce| displays it.
```
Comment (JuliaMongo) [inline source/includes/seq-edit-document.rst:7]: Can we reverse these unordered bullets? Success first, then error?

File: source/read-document-playground.txt [hunk 8]
```diff
@@ -61,6 +61,10 @@ in the MongoDB Manual.
 
 .. include:: /includes/run-playground.rst
 
+You may edit the returned |json| document.
```
Comment (JuliaMongo) [inline source/read-document-playground.txt]: Can this be, at least:
While you are viewing one document, you can also edit it. 
Then add a stem sentence to the include procedure. As written, this is a procedure without its own heading or stem sentence, tacked onto another topic, and it is confusing. Maybe, the title can be changed to : Read one and edit? Something's off here and if this section is what you added, and it is in scope, consider one of these:
- changing the title to indicate that edits are also possible
- saying: while you view, you can also edit
- adding a stem sentence before the included steps for how to edit.

File: source/includes/seq-edit-document.rst (hunk 7 above)
Comment (atsansone) [reply source/includes/seq-edit-document.rst:7]: Agreed. Fixed.

File: source/read-document-playground.txt (hunk 8 above)
Comment (atsansone) [reply source/read-document-playground.txt]: This was updated to: `You may edit any |json| document returned from a |findOne| or |findAny| operation.`.

File: source/databases-collections.txt (hunk 6 above)
Comment (JuliaMongo) [reply source/databases-collections.txt:20]: Agreed that we shouldn't fix there. I will add this to the ticket I opened.

**Comment by JuliaMongo** [comment]
LGTM!

**Review by JuliaMongo — APPROVED** [review]
LGTM

**Pull request merged**